Minesweeper.py

```diff
@@ -1,40 +1,66 @@
 import pygame
 import random
 
-# DHB: it is somewhat unnatural to see pygame used before being initialised. Also, it is not cool to execute code in multiple parts
-# DHB: of the file. Define your classes and functions first, then execute your code.
-# Create the game window
-win = pygame.display.set_mode((888, 938))
-pygame.display.set_caption("Minesweeper")
-
-# DHB: all classes in python should contain class comments, rather than preceeding the class with a comment like this
-# Define the Color class to store color values
 class Color(object):
-    # DHB: it is not great to skip the parenthesis around a tuple, even though it is legal. If you would e.g. copy this constant
-    # DHB: elsewhere, such as in a function parameter list, then it would change from being a tuple to being multiple parameters 
-    GRAY = 155, 155, 155
-    BACKGROUND = 120, 120, 120
-    BLACK = 0, 0, 0
-    WHITE = 255, 255, 255
-    DARK_GRAY = 100, 100, 100
-    RED = 255, 0, 0
-    BLUE = 0, 0, 255
-    GREEN = 0, 255, 0
-    DARK_BLUE = 0, 0, 139
-    MAROON = 128, 0, 0
-    TURQUOISE = 48, 213, 200
-
-# DHB: incorrect approach to commenting a class
-class TutorialScreen(object): # creates the tutorial screen
+    """ store color values in constants to not use harcoded values later
+    """
+    GRAY = (155, 155, 155)
+    BACKGROUND = (120, 120, 120)
+    BLACK = (0, 0, 0)
+    WHITE = (255, 255, 255)
+    DARK_GRAY = (100, 100, 100)
+    RED = (255, 0, 0)
+    BLUE = (0, 0, 255)
+    GREEN = (0, 255, 0)
+    DARK_BLUE = (0, 0, 139)
+    MAROON = (128, 0, 0)
+    TURQUOISE = (48, 213, 200)
+
+class GameState(object):
+    """ changes the game states which are used in the code to fulfill conditions"""
     def __init__(self):
-        # DHB: missing function comments in this, and most other functions in this file.
-        # DHB: comments should be in the preceeding line, not as a trailing comment
-        # DHB: also, when parameters are not obvious, it is wise to call them by name rather than position
-        # DHB: e.g. Font(file_path=None, size=30) 
-        self.font = pygame.font.Font(None, 30) # Font for tutorial text
-        # DHB: excellent choice to use named constants instead of hardcoded values
-        self.text_color = Color.BLACK # Color for tutorial text
-        self.instructions = [
+        """sets the states to their beginning values, which can then be chaged later"""
+        self.tutorial = True
+        self.game = False
+        self.game_over = False
+        self.game_won = False
+
+    def set_state_tutorial(self):
+        if self.tutorial == True:
+            self.tutorial = False
+        elif self.tutorial == False:
```

Comment:
since self.tutorial is a boolean, it can only be True or False. Thus it is better to write this as else: